tools/typed_ast.c

```diff
@@ -0,0 +1,329 @@
+#include "Python.h"
```

Comment:
What is putting this into `tools` for?

Reply:
Oh, you should probably ignore everything I did to tools. This was an attempt at automating the update process more, but it's really not working very well yet, and I should just redo that if we even need it.

If all goes as I hope, Python 3.8 will support type comments natively, and then extracting the upstream ast module into typed_ast will be much simpler (though we'll still need to do the work to backport the code to earlier CPython versions and C89). Even if this doesn't happen, we should probably keep a branch of CPython live and only do purely mechanical transformations from there to typed_ast -- merging upstream into that branch should be simpler that the current update process.